In [1]:
import pandas as pd

RAW_PATH = "../data/raw/"

jobs_path = RAW_PATH + "job_postings.csv"
skills_path = RAW_PATH + "job_skills.csv"
summary_path = RAW_PATH + "job_summary.csv"

In [2]:
df_jobs = pd.read_csv(jobs_path)

print("CSV shape: ", df_jobs.shape)
print("\nNulls:\n", df_jobs.isnull().sum())
print("\nUnique job_link: ", df_jobs["job_link"].nunique())
df_jobs.head(3)

CSV shape:  (12217, 15)

Nulls:
 job_link               0
last_processed_time    0
last_status            0
got_summary            0
got_ner                0
is_being_worked        0
job_title              0
company                0
job_location           1
first_seen             0
search_city            0
search_country         0
search_position        0
job_level              0
job_type               0
dtype: int64

Unique job_link:  12217


,job_link,last_processed_time,last_status,got_summary,got_ner,is_being_worked,job_title,company,job_location,first_seen,search_city,search_country,search_position,job_level,job_type
0,https://www.linkedin.com/jobs/view/senior-mach...,2024-01-21 08:08:48.031964+00,Finished NER,t,t,f,Senior Machine Learning Engineer,Jobs for Humanity,"New Haven, CT",2024-01-14,East Haven,United States,Agricultural-Research Engineer,Mid senior,Onsite
1,https://www.linkedin.com/jobs/view/principal-s...,2024-01-20 04:02:12.331406+00,Finished NER,t,t,f,"Principal Software Engineer, ML Accelerators",Aurora,"San Francisco, CA",2024-01-14,El Cerrito,United States,Set-Key Driver,Mid senior,Onsite
2,https://www.linkedin.com/jobs/view/senior-etl-...,2024-01-21 08:08:31.941595+00,Finished NER,t,t,f,Senior ETL Data Warehouse Specialist,Adame Services LLC,"New York, NY",2024-01-14,Middletown,United States,Technical Support Specialist,Associate,Onsite


In [3]:
df_skills = pd.read_csv(skills_path)

print("CSV shape: ", df_skills.shape)
print("\nNulls:\n", df_skills.isnull().sum())
print("\nUnique job_link: ", df_skills["job_link"].nunique())
df_skills.head(3)

CSV shape:  (12217, 2)

Nulls:
 job_link      0
job_skills    5
dtype: int64

Unique job_link:  12217


,job_link,job_skills
0,https://www.linkedin.com/jobs/view/senior-mach...,"Machine Learning, Programming, Python, Scala, ..."
1,https://www.linkedin.com/jobs/view/principal-s...,"C++, Python, PyTorch, TensorFlow, MXNet, CUDA,..."
2,https://www.linkedin.com/jobs/view/senior-etl-...,"ETL, Data Integration, Data Transformation, Da..."


In [4]:
df_summary = pd.read_csv(summary_path)

print("CSV shape: ", df_summary.shape)
print("\nNulls:\n", df_summary.isnull().sum())
print("\nUnique job_link: ", df_summary["job_link"].nunique())
df_skills.head(3)

CSV shape:  (12217, 2)

Nulls:
 job_link       0
job_summary    0
dtype: int64

Unique job_link:  12217


,job_link,job_skills
0,https://www.linkedin.com/jobs/view/senior-mach...,"Machine Learning, Programming, Python, Scala, ..."
1,https://www.linkedin.com/jobs/view/principal-s...,"C++, Python, PyTorch, TensorFlow, MXNet, CUDA,..."
2,https://www.linkedin.com/jobs/view/senior-etl-...,"ETL, Data Integration, Data Transformation, Da..."


In [5]:
df_jobs.info()

<class 'pandas.DataFrame'>
RangeIndex: 12217 entries, 0 to 12216
Data columns (total 15 columns):
 #   Column               Non-Null Count  Dtype
---  ------               --------------  -----
 0   job_link             12217 non-null  str  
 1   last_processed_time  12217 non-null  str  
 2   last_status          12217 non-null  str  
 3   got_summary          12217 non-null  str  
 4   got_ner              12217 non-null  str  
 5   is_being_worked      12217 non-null  str  
 6   job_title            12217 non-null  str  
 7   company              12217 non-null  str  
 8   job_location         12216 non-null  str  
 9   first_seen           12217 non-null  str  
 10  search_city          12217 non-null  str  
 11  search_country       12217 non-null  str  
 12  search_position      12217 non-null  str  
 13  job_level            12217 non-null  str  
 14  job_type             12217 non-null  str  
dtypes: str(15)
memory usage: 1.4 MB


In [6]:
df_skills.info()

<class 'pandas.DataFrame'>
RangeIndex: 12217 entries, 0 to 12216
Data columns (total 2 columns):
 #   Column      Non-Null Count  Dtype
---  ------      --------------  -----
 0   job_link    12217 non-null  str  
 1   job_skills  12212 non-null  str  
dtypes: str(2)
memory usage: 191.0 KB


In [7]:
df_summary.info()

<class 'pandas.DataFrame'>
RangeIndex: 12217 entries, 0 to 12216
Data columns (total 2 columns):
 #   Column       Non-Null Count  Dtype
---  ------       --------------  -----
 0   job_link     12217 non-null  str  
 1   job_summary  12217 non-null  str  
dtypes: str(2)
memory usage: 191.0 KB


In [8]:
for col in ["job_title", "job_level", "job_type", "search_country"]:
    print(f"\n{col} ({df_jobs[col].nunique()} únicos):")
    print(df_jobs[col].value_counts().head(10))


job_title (6484 únicos):
job_title
Senior Data Engineer                         285
Senior Data Analyst                          163
Data Engineer                                149
Senior MLOps Engineer                        138
Data Analyst                                 137
Data Scientist                               128
Lead Data Engineer                           123
Senior Data Scientist                        119
Data Architect                               111
Staff Machine Learning Engineer, Series A    101
Name: count, dtype: int64

job_level (2 únicos):
job_level
Mid senior    10919
Associate      1298
Name: count, dtype: int64

job_type (3 únicos):
job_type
Onsite    12188
Remote       18
Hybrid       11
Name: count, dtype: int64

search_country (4 únicos):
search_country
United States     10291
United Kingdom      995
Canada              630
Australia           301
Name: count, dtype: int64


In [9]:
df_skills["job_skills"].head(10)

0    Machine Learning, Programming, Python, Scala, ...
1    C++, Python, PyTorch, TensorFlow, MXNet, CUDA,...
2    ETL, Data Integration, Data Transformation, Da...
3    Data Lakes, Data Bricks, Azure Data Factory Pi...
4    Java, Scala, Python, RDBMS, NoSQL, Redshift, S...
5    Data Warehouse (DW), Extract/Transform/Load (E...
6    Machine Learning, Generative AI, Cloudbased se...
7    Data Loss Prevention (DLP), Cybersecurity, Web...
8    Problem solving, Data analysis, Predictive ana...
9    Machine Learning, System Design, Multimodal In...
Name: job_skills, dtype: str

##Exploration conclusions

- 12217 records across the 3 files, with a clean JOIN using they key: job_link
- Almost any missing values.
- Key job_skills requires parsing (split by comma + explode)
- job_title requires normalization to group similar roles
- Dataset is predominantly U.S.-based (84%) and Onsite (99%)
